pkg/pprof/merge_test.go

```diff
@@ -17,6 +20,29 @@ func Test_Merge_Single(t *testing.T) {
 	testhelper.EqualProto(t, p.Profile, m.Profile())
 }
 
+func Fuzz_Merge_Single(f *testing.F) {
+	p, err := OpenFile("testdata/go.cpu.labels.pprof")
+	require.NoError(f, err)
+
+	buf := new(bytes.Buffer)
+	require.NoError(f, gob.NewEncoder(buf).Encode(p.Profile))
+	f.Add(buf.Bytes())
+
+	f.Fuzz(func(t *testing.T, data []byte) {
+		var p Profile
+		err := gob.NewDecoder(bytes.NewReader(data)).Decode(&p.Profile)
+		if err != nil {
+			return
+		}
+
+		var m ProfileMerge
+		err = m.Merge(p.Profile)
+		if err != nil {
+			return
+		}
+	})
+}
+
```

Comment:
What do you think about using use of proto encoding? It's way faster and seems more natural

```go
func Fuzz_Merge_Single(f *testing.F) {
	raw, err := OpenFile("testdata/go.cpu.labels.pprof")
	require.NoError(f, err)
	data, err := raw.MarshalVT()
	require.NoError(f, err)
	f.Add(data)

	f.Fuzz(func(t *testing.T, data []byte) {
		var p profilev1.Profile
		if err = p.UnmarshalVT(data); err != nil {
			var m ProfileMerge
			_ = m.Merge(&p)
		}
	})
}
```

Reply:
I think I was having trouble because I forgot about gzip, when doing an approach like that 🙂